crates/sui-graphql-rpc/src/types/suins_registration.rs

```diff
@@ -310,37 +327,99 @@ impl SuinsRegistration {
 impl NameService {
     /// Lookup the SuiNS NameRecord for the given `domain` name. `config` specifies where to find
     /// the domain name registry, and its type.
+    ///
+    /// `checkpoint_viewed_at` represents the checkpoint sequence number at which this was queried
+    /// for, or `None` if the data was requested at the latest checkpoint.
+    ///
+    /// The `NameRecord` is returned only if it has not expired as of the `checkpoint_viewed_at` or
+    /// latest checkpoint's timestamp.
+    ///
+    /// For leaf domains, the `NameRecord` is returned only if its parent is valid and not expired.
     pub(crate) async fn resolve_to_record(
-        db: &Db,
-        config: &NameServiceConfig,
+        ctx: &Context<'_>,
         domain: &Domain,
+        checkpoint_viewed_at: Option<u64>,
     ) -> Result<Option<NameRecord>, Error> {
-        let record_id = config.record_field_id(&domain.0);
+        let config = ctx.data_unchecked::<NameServiceConfig>();
+        // Create a page with a bound of 2 to fetch the domain's NameRecord and potentially its parent's
+        let page: Page<object::Cursor> =
+            Page::from_params(ctx.data_unchecked(), Some(2), None, None, None).map_err(|_| {
+                Error::Internal("Page size of 2 is incompatible with configured limits".to_string())
+            })?;
 
-        let Some(object) = MoveObject::query(db, record_id.into(), ObjectLookupKey::Latest).await?
+        let record_id: SuiAddress = config.record_field_id(&domain.0).into();
+        let parent_record_id: SuiAddress = config.record_field_id(&domain.0.parent()).into();
+
+        // Query for the domain's NameRecord and parent NameRecord if applicable. The checkpoint's
+        // timestamp is also fetched. These values are used to determine if the domain is expired.
+        let Some(query) = Self::query_domain_expiration(
```

Comment:
nit: `query` here represents the result of the query and not the query itself right? So it's really the `domain_expiration` rather than the `query`.

Reply:
yes makes sense